src/types.ts

```diff
@@ -53,6 +81,86 @@ export interface FrameStartType {
   sceneFrameCounter: number | null;
 }
 
+export interface GameInfoType {
+  gameBitfield1: number | null;
+  gameBitfield2: number | null;
+  gameBitfield3: number | null;
+  gameBitfield4: number | null;
+  bombRainEnabled: boolean | null;
+  itemSpawnBehavior: number | null;
+  selfDestructScoreValue: number | null;
+  startingTimerFrames: number | null;
+  itemSpawnBitfield1: number | null;
+  itemSpawnBitfield2: number | null;
+  itemSpawnBitfield3: number | null;
+  itemSpawnBitfield4: number | null;
+  itemSpawnBitfield5: number | null;
+  damageRatio: number | null;
+}
+
+export enum TimerType {
+  NONE = 0b00,
+  DECREASING = 0b10,
+  INCREASING = 0b11,
+}
+
+export enum ItemSpawnBehaviorType {
+  OFF = 0xff,
+  VERY_LOW = 0x00,
+  LOW = 0x01,
+  MEDIUM = 0x02,
+  HIGH = 0x03,
+  VERY_HIGH = 0x04,
+}
+
+export enum EnabledItemType {
+  METAL_BOX = 2 ** 0,
+  CLOAKING_DEVICE = 2 ** 1,
+  POKEBALL = 2 ** 2,
+  // Bits 4 through 8 of item bitfield 1 are unknown
+  UNKNOWN_ITEM_BIT_4 = 2 ** 3,
+  UNKNOWN_ITEM_BIT_5 = 2 ** 4,
+  UNKNOWN_ITEM_BIT_6 = 2 ** 5,
+  UNKNOWN_ITEM_BIT_7 = 2 ** 6,
+  UNKNOWN_ITEM_BIT_8 = 2 ** 7,
+  FAN = 2 ** 8,
+  FIRE_FLOWER = 2 ** 9,
+  SUPER_MUSHROOM = 2 ** 10,
+  POISON_MUSHROOM = 2 ** 11,
+  HAMMER = 2 ** 12,
+  WARP_STAR = 2 ** 13,
+  SCREW_ATTACK = 2 ** 14,
+  BUNNY_HOOD = 2 ** 15,
+  RAY_GUN = 2 ** 16,
+  FREEZIE = 2 ** 17,
+  FOOD = 2 ** 18,
+  MOTION_SENSOR_BOMB = 2 ** 19,
+  FLIPPER = 2 ** 20,
+  SUPER_SCOPE = 2 ** 21,
+  STAR_ROD = 2 ** 22,
+  LIPS_STICK = 2 ** 23,
+  HEART_CONTAINER = 2 ** 24,
+  MAXIM_TOMATO = 2 ** 25,
+  STARMAN = 2 ** 26,
+  HOME_RUN_BAT = 2 ** 27,
+  BEAM_SWORD = 2 ** 28,
+  PARASOL = 2 ** 29,
+  GREEN_SHELL = 2 ** 30,
+  RED_SHELL = 2 ** 31,
+  CAPSULE = 2 ** 32,
+  BOX = 2 ** 33,
+  BARREL = 2 ** 34,
+  EGG = 2 ** 35,
+  PARTY_BALL = 2 ** 36,
+  BARREL_CANNON = 2 ** 37,
+  BOMB_OMB = 2 ** 38,
+  MR_SATURN = 2 ** 39,
+}
+
+export interface EnabledItemsType {
+  items: string[] | null;
```

Comment:
```suggestion
  items: string[] | null;
```
Why is this an array of strings? Also why do we even need this type? It seems kinda useless. Can't we just return `EnabledItemType[]`?